src/ffi.rs

```diff
@@ -18,7 +18,7 @@ macro_rules! ffi_buf {
             use std::mem;
 
             let mut buf = unsafe { vec![mem::zeroed(); $sz as usize] };
-            *(&mut $ptr) = unsafe { mem::transmute(buf.as_mut_ptr()) };
+            *(&mut $ptr) = unsafe { buf.as_mut_ptr() as u64};
```

Comment:
On 32-bit targets, this will not compile, since the destination type is 64-bits, while `buf.as_mut_ptr()` is 32-bits. If there is a way to do this that doesn't sacrifice the generality of the macro, I will happily change it. In practice, this macro is only ever used to populate 64-bit ingegers.

Reply:
Makes sense. Might not even need the unsafe at this point, but I'll look at that later.